crypto/cmp/cmp_client.c

```diff
@@ -741,11 +754,20 @@ X509 *OSSL_CMP_exec_certreq(OSSL_CMP_CTX *ctx, int req_type,
     int rid = is_p10 ? OSSL_CMP_CERTREQID_NONE : OSSL_CMP_CERTREQID;
     int rep_type = is_p10 ? OSSL_CMP_PKIBODY_CP : req_type + 1;
     X509 *result = NULL;
+    int kembasedmac;
 
     if (ctx == NULL) {
         ERR_raise(ERR_LIB_CMP, CMP_R_NULL_ARGUMENT);
         return NULL;
     }
+    kembasedmac = ossl_cmp_kem_BasedMac_required(ctx);
+    if (kembasedmac == -1) {
+        goto err;
+    } else if (kembasedmac == 1
+               && ctx->kem == KBM_SSK_USING_CLINET_KEM_KEY) {
+        if (!OSSL_CMP_get_ssk(ctx))
+            goto err;
+    }
```

Comment:
Why is this done only for cert requests?
Should be at the start of any type of transaction!